building/qubes-r3-building.md

```diff
@@ -24,36 +38,80 @@ $ wget https://keys.qubes-os.org/keys/qubes-master-signing-key.asc
 $ gpg --import qubes-master-signing-key.asc 
 $ gpg --edit-key 36879494
 # Verify fingerprint!, set trust to *ultimate*
+# (Typical sequence is fpr, trust, 5, q)
 $ wget https://keys.qubes-os.org/keys/qubes-developers-keys.asc
 $ gpg --import qubes-developers-keys.asc
 ~~~
 
-Note we do *not* relay above on the security of our server (keys.qubes-os.org) nor the connection (ssl, cert) -- we only rely on you getting the Qubes Master Signing Key fingerprint *somehow* and ensure they match!
+And if you will be building the Whonix templates:
+
+~~~
+$ wget https://github.com/QubesOS/qubes-builder-debian/blob/master/keys/whonix-developer-patrick.asc
+$ gpg --import whonix-developer-patrick.asc
+~~~
+
+**Note** In the above process, we do *not* rely on the security of our server (keys.qubes-os.org) nor the connection (ssl, cert) -- we only rely on you getting the Qubes Master Signing Key fingerprint *somehow* and ensuring they match!
+Likewise, the Whonix signing key is available from multiple sources.
 
-Now lets bootstrap the builder. Unfortunately the builder cannot verify itself (the classic Chicken and Egg problem), so we need to verify the signature manually:
+Now let's bootstrap the builder. Unfortunately, the builder cannot verify itself (the classic Chicken and Egg problem), so we need to verify the signature manually:
 
 ~~~
 $ git clone git://github.com/QubesOS/qubes-builder.git
+$ mkdir qubes-builder/keyrings
+$ mkdir qubes-builder/keyrings/git
+$ cp .gnupg/pubring.gpg qubes-builder/keyrings/git/
+$ cp .gnupg/trustdb.gpg qubes-builder/keyrings/git/
 $ cd qubes-builder
-$ git describe --exact-match HEAD
-<some tag>
-$ git tag -v <some tag>
+$ git tag -v `git describe`
+~~~
```

Comment:
This whole step can be simplified to the old version: you don't need to use `qubes-builder/keyrings/git/pubring.gpg` for manual verification. After all, you've imported master signing key to your default keyring, not here.
Instead, I'd add extra step:
```
gpg --import qubes-developers-keys.asc
```
This is because tags signatures on qubes-builder (and every other component) are made using some of those keys. And those keys are signed by the master key. Since the master key is already set to trust 'ultimate', gpg will use that when evaluating trustworthiness of the keys here. It's very important to check if the verification message contains "Good signature from ..." and _do not contains_ "WARNING: This key is not certified with a trusted signature!".

Reply:
`gpg --import qubes-developers-keys.asc` is already done in the previous section. Are you saying this one can be simplified to:
```
$ git clone git://github.com/QubesOS/qubes-builder.git
$ cd qubes-builder
$ git tag -v `git describe`
```
?

Reply:
Yes. Plus info how to interpret result - see previous comment.

Reply:
Ah, you've already added it :)